diff --git a/typedoc/src/lib/output/plugins/JavascriptIndexPlugin.ts b/packages/plugin-pages/src/theme-plugins/search/default-pages-javascript-index-plugin.GENERATED.ts
index 454d8c1..180d365 100644
--- a/typedoc/src/lib/output/plugins/JavascriptIndexPlugin.ts
+++ b/packages/plugin-pages/src/theme-plugins/search/default-pages-javascript-index-plugin.GENERATED.ts
@@ -1,29 +1,64 @@
+// Edit of <~/typedoc/src/lib/output/plugins/JavascriptIndexPlugin.ts>
+import assert from 'assert';
+import { mkdirSync, writeFileSync } from 'fs';
 import * as Path from 'path';
+
+import { isString } from 'lodash';
 import { Builder, trimmer } from 'lunr';
+import { DeclarationReflection, ProjectReflection, Reflection, ReflectionKind, RendererEvent } from 'typedoc';
+import type { GroupPlugin } from 'typedoc/dist/lib/converter/plugins';
+import { JavascriptIndexPlugin } from 'typedoc/dist/lib/output/plugins';
 
-import {
-	DeclarationReflection,
-	ProjectReflection,
-	ReflectionKind,
-} from '../../models';
-import { GroupPlugin } from '../../converter/plugins';
-import { Component, RendererComponent } from '../components';
-import { RendererEvent } from '../events';
-import { writeFileSync } from '../../utils';
-import { DefaultTheme } from '../themes/default/DefaultTheme';
+import type { PagesPlugin } from '../../plugin';
+import { ANodeReflection, MenuReflection } from '../../reflections';
 
 /**
  * A plugin that exports an index of the project to a javascript file.
  *
  * The resulting javascript file can be used to build a simple search function.
  */
-@Component( { name: 'javascript-index' } )
-export class JavascriptIndexPlugin extends RendererComponent {
+export class DefaultPagesJavascriptIndexPlugin {
+	private readonly _defaultSearch: JavascriptIndexPlugin;
+	private readonly _groupPlugin: GroupPlugin;
+	public constructor( private readonly _plugin: PagesPlugin ) {
+		this._groupPlugin = this._plugin.application.converter.getComponent( 'group' ) as GroupPlugin;
+		this._defaultSearch = this._plugin.application.renderer.getComponent( 'javascript-index' ) as any as JavascriptIndexPlugin;
+	}
 	/**
 	 * Create a new JavascriptIndexPlugin instance.
 	 */
-	override initialize() {
-		this.listenTo( this.owner, RendererEvent.BEGIN, this.onRendererBegin );
+	public initialize(){
+		// eslint-disable-next-line @typescript-eslint/dot-notation -- Private property
+		const fn = this._defaultSearch['onRendererBegin'];
+		assert( fn );
+		this._plugin.application.renderer.off( RendererEvent.BEGIN, fn );
+		this._plugin.application.renderer.on( RendererEvent.BEGIN, this._onRendererBegin.bind( this ) );
+	}
+
+	/**
+	 * Convert {@link ANodeReflection} instances to custom declaration reflections with specific properties set.
+	 *
+	 * @param reflection - The reflection to convert.
+	 * @param project - The root project, used to get modules.
+	 * @returns the new reflection, along with a flag indicating if it is a page.
+	 */
+	private _convertNodeReflection( reflection: Reflection, project: ProjectReflection ){
+		if( !( reflection instanceof ANodeReflection ) ){
+			return { reflection, isPage: false };
+		}
+		if( !this._plugin.pluginOptions.getValue().enableSearch || reflection instanceof MenuReflection ){
+			return { reflection: null, isPage: false };
+		}
+		const name = [
+			'Page:',
+			reflection.module !== project ? `${reflection.module.name} ⇒` : undefined,
+			reflection.name,
+		].filter( isString ).join( ' ' );
+		const dec = new DeclarationReflection( name, ReflectionKind.Method );
+		dec.comment = reflection.comment;
+		dec.url = reflection.url;
+		dec.cssClasses = 'tsd-kind-method tsd-parent-kind-interface tsd-is-inherited tsd-is-external pages-entry';
+		return { reflection: dec, isPage: true };
 	}
 
 	/**
@@ -31,34 +66,34 @@ export class JavascriptIndexPlugin extends RendererComponent {
 	 *
 	 * @param event  - An event object describing the current render operation.
 	 */
-	private onRendererBegin( event: RendererEvent ) {
-		if ( !( this.owner.theme instanceof DefaultTheme ) ) {
-			return;
-		}
+	private _onRendererBegin( event: RendererEvent ) {
 		if ( event.isDefaultPrevented ) {
 			return;
 		}
 
+		this._plugin.logger.verbose( `${this._plugin.pluginOptions.getValue().enableSearch ? 'Enabling' : 'Disabling'} search for pages` );
 		const rows: any[] = [];
 		const kinds: { [K in ReflectionKind]?: string } = {};
 
 		const kindBoosts =
-            ( this.application.options.getValue( 'searchGroupBoosts' ) as {
-            	[key: string]: number;
-            } ) ?? {};
-
-		for ( const reflection of event.project.getReflectionsByKind(
-			ReflectionKind.All
-		) ) {
+			( this._plugin.application.options.getValue( 'searchGroupBoosts' ) as {
+				[key: string]: number;
+			} ) ?? {};
+
+		const reflections = event.project.getReflectionsByKind( ReflectionKind.All )
+			.map( r => this._convertNodeReflection( r, event.project ) );
+		const pagesCount = reflections.filter( r => r.isPage ).length;
+		this._plugin.logger.verbose( `Adding ${pagesCount} pages to the search index` );
+		for ( const { reflection } of reflections ) {
 			if ( !( reflection instanceof DeclarationReflection ) ) {
 				continue;
 			}
 
 			if (
 				!reflection.url ||
-                !reflection.name ||
-                reflection.flags.isExternal ||
-                reflection.name === ''
+				!reflection.name ||
+				reflection.flags.isExternal ||
+				reflection.name === ''
 			) {
 				continue;
 			}
@@ -70,14 +105,14 @@ export class JavascriptIndexPlugin extends RendererComponent {
 			}
 
 			if ( !kinds[reflection.kind] ) {
-				kinds[reflection.kind] = GroupPlugin.getKindSingular(
-					reflection.kind
+				kinds[reflection.kind] = ( this._groupPlugin.constructor as typeof GroupPlugin ).getKindSingular(
+					reflection.kind,
 				);
-			}
 
-			const kindBoost = kindBoosts[kinds[reflection.kind] ?? ''];
-			if ( kindBoost != undefined ) {
-				boost *= kindBoost;
+				const kindBoost = kindBoosts[kinds[reflection.kind] ?? ''];
+				if ( kindBoost !== undefined ) {
+					boost *= kindBoost;
+				}
 			}
 
 			const row: any = {
@@ -113,7 +148,7 @@ export class JavascriptIndexPlugin extends RendererComponent {
 		const jsonFileName = Path.join(
 			event.outputDirectory,
 			'assets',
-			'search.js'
+			'search.js',
 		);
 
 		const jsonData = JSON.stringify( {
@@ -122,9 +157,10 @@ export class JavascriptIndexPlugin extends RendererComponent {
 			index,
 		} );
 
+		mkdirSync( Path.dirname( jsonFileName ), { recursive: true } );
 		writeFileSync(
 			jsonFileName,
-			`window.searchData = JSON.parse(${JSON.stringify( jsonData )});`
+			`window.searchData = JSON.parse(${JSON.stringify( jsonData )});`,
 		);
 	}
 }
